File: DataStructures/Queue/Queue.py
#Queue implementation using Linked list

class Node: 
    def __init__(self, data): 
        self.data = data 
        self.next = None
  
class Queue: 
      
    def __init__(self): 
        self.front = None
        self.rear = None
    
    def isEmpty(self): 
        return self.front == None and self.rear == None

    #add item to queue  
    def enqueue(self, item): 
        newNode = Node(item) 
          
        if self.rear == None: 
            self.front = self.rear = newNode 
            return
        self.rear.next = newNode 
        self.rear = newNode 
  
    #remove first item of queue 
    def dequeue(self): 
        if not self.isEmpty(): 
            node = self.front 
            self.front = self.front.next
    
            if(self.front == None): 
                self.rear = None
                
            return node
    #get first item of queue without remove
    def peek(self):
        if(not self.isEmpty()):
            return self.front.data
        return None

File: DataStructures/Queue/Queue.java
//Queue implementation using Linked list

class Node {
    int data;
    Node next;

    public Node(int data){
        this.data = data;
        this.next = null;
    }
}

class Queue {
    Node front;
    Node rear;

    public Queue(){
        this.front = null;
        this.rear = null;
    }

    //Add item to queue 
    public void enqueue(int data){
        Node newNode = new Node(data);

        //If queue is empty
        if (this.rear == null) { 
            this.front = newNode;
            this.rear = newNode; 
            return; 
        } 
        //Change last node next to newNode
        this.rear.next = newNode; 
        //Change rear to newNode
        this.rear = newNode; 
    }

    //Remove first item of queue 
    public Node dequeue(){
        if(this.front != null){
            Node node = this.front;

            this.front = this.front.next;

            if(this.front == null)
                this.rear = null;
            
            return node;
        }
        return null;
    }

    //Get first item of queue without remove
    public Node peek(){
        return this.front;
    }

    public boolean isEmpty(){
        return front == null && rear == null;
    }

    void print(){
        if(!isEmpty()){
            Node current = front;
            while(current != null){
                System.out.print(current.data + " <-- ");
                current = current.next;
            }
            System.out.println();
        }
        else
            System.out.println("Queue is empty");
    }

}